### DISCUSSION TITLE
3D coordinates on sulcis. BrainVISA

### DISCUSSION THREAD

**Discussion by eherrelo11** [brainvisa/brainvisa.github.io#45] [Q&A]
Good morning,
 

I would like to know if there is a tool in BrainVISA to automatically place landmarks (3D coordinates) at the starting/termination or intersection points of the sulci. Since there is an option of sulcus parameterization, which offers a definition of a coordinate system on the sulcus, would it be possible to obtain an output with these coordinates in an automatic manner for the different sulci?

The aim is to obtain an overall characterization of cortical morphology based on a limited number of landmarks in order to perform geometric morphometric analyses and to study morphological shape variation among individuals. By doing it automatically, using sulcus parameterization information, we would avoid inter/intra-observer error derived from the manual placement of the landmarks.

Thank you very much.

Elena.

**Comment by denisri** [comment]
Hi,
The sulcus parameterization process works for sulci with a "linear" shape, it is not designed for sulci with complex shapes, intersections or interruptions, thus it is probably not what you are looking for.
The oldest sulci recognition system (ANN) and it associated morphometry process was recording the extremity points of all sulci (the points located furthest through geodesic distances). You can try that. I just realize that the morphometry process crashes when exiting (in the program cleaining parts at exit), but it seems to have finished its job at this point.
Denis

**Comment by Mhaalk** [comment]
Hi Elena,

some leads maybe:
to get 3D coordinates of beginnings/ends of sulci, my guess would be to run a script on individual sulci .nii files (which are basically 3D arrays with a header) that grab the maximum coordinates on each axis where there is a non null value.

to get intersections: 
either 2 sulci arrays indeed share a non null value on the same x/y/z coordinates (I don't think it happens a lot), in which case adding values from 2 arrays will spot intersections where the array value is doubled; 
either they visually "touch" but never overlap, in which case a scalable/reproducible way to point landmarks would be to calculate distance between non null values of each sulci arrays, and point minimal distances below a defined threshold.

aims functions allow to do this kind of things with python, hope I made myself clear.

M
